### ISSUE TITLE
Report variables that not being assigned any value at all.

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#772]
**Is your feature request related to a problem? Please describe.**
The current version of Pyright won't report unassigned variables as long as they have a type annotation.
Please see the screenshot.
![image](https://user-images.githubusercontent.com/3479453/85948826-7d2b4c80-b985-11ea-8bff-9f3a9cd18798.png)
![image](https://user-images.githubusercontent.com/3479453/85948813-6a187c80-b985-11ea-871c-d6f81a4f9fcc.png)

Variable `a` in `fn1` and variable `b` in `fn2` can both lead to runtime error `local variable X referenced before assignment`, but only the second one is being reported by Pyright now.

**Describe the solution you'd like**
I think `a` in `fn1` should also be reported as an error (not sure what kind of error though) because there will be a runtime error.

**Additional context**
Thanks!

Codes:
```python
def fn1():
    a: int
    return a

fn1()

def fn2():
    b
    return b

fn2()
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in Pyright version 1.1.48, which I just published. It will also be included in the next public release of Pylance.
